File: app/src/main/java/com/example/asthmaapplication/main/mainpage/subpages/PatientsFragment.java
package com.example.asthmaapplication.main.mainpage.subpages;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProvider;

import com.example.asthmaapplication.R;
import com.example.asthmaapplication.databinding.FragmentPatientsBinding;
import com.example.asthmaapplication.main.common.BaseFragment;
import com.example.asthmaapplication.main.mainpage.MainViewModel;

import javax.inject.Inject;

import io.reactivex.annotations.NonNull;

public class PatientsFragment extends BaseFragment {
    MainViewModel viewModel;
    FragmentPatientsBinding binding;

    @Inject
    public PatientsFragment() {

    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewModel = new ViewModelProvider(this).get(MainViewModel.class);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @javax.annotation.Nullable ViewGroup container, @javax.annotation.Nullable Bundle savedInstanceState) {
        binding = FragmentPatientsBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@Nullable View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        getPreferences();
        setActionBarTitle();
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    public void setActionBarTitle() {
        setActionBarTitle(getString(R.string.patients_page_title, getUserName()));
    }

    @Override
    public View getRoot() {
        return binding.getRoot();
    }
}

File: app/src/main/java/com/example/asthmaapplication/main/mainpage/subpages/learningbasepages/PatientsFragment.kt
package com.example.asthmaapplication.main.mainpage.subpages.learningbasepages

import javax.inject.Inject
import com.example.asthmaapplication.main.common.BaseFragment
import com.example.asthmaapplication.main.mainpage.MainViewModel
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.asthmaapplication.databinding.FragmentPatientsBinding
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class PatientsFragment @Inject constructor() : BaseFragment() {
    var viewModel: MainViewModel? = null
    var binding: FragmentPatientsBinding? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentPatientsBinding.inflate(inflater, container, false)
        return binding!!.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view!!, savedInstanceState)
        setActionBarTitle()
    }

    override fun onResume() {
        super.onResume()
    }

    fun setActionBarTitle() {}
    override fun getRoot(): View {
        return binding!!.root
    }
}